src/dds/rs-dds-internal-data.cpp

```diff
@@ -0,0 +1,268 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include "rs-dds-internal-data.h"
+
+#include <librealsense2/h/rs_sensor.h>
+#include <proc/processing-blocks-factory.h>
+#include <proc/color-formats-converter.h>
+#include <proc/depth-formats-converter.h>
+#include <proc/y8i-to-y8y8.h>
+#include <proc/y12i-to-y16y16.h>
+#include <proc/y12i-to-y16y16-mipi.h>
+
+namespace librealsense {
+
+static const std::string RS405_PID = "0B5B";
+static const std::string RS415_PID = "0AD3";
+static const std::string RS435_RGB_PID = "0B07";
+static const std::string RS435I_PID = "0B3A";
+static const std::string RS455_PID = "0B5C";
+static const std::string RS457_PID = "ABCD";
+
+
+std::vector< rs2_format > target_formats( rs2_format source_format )
+{
+    // Mapping from source color format to all of the compatible target color formats.
+    std::vector< rs2_format > formats = { RS2_FORMAT_RGB8, RS2_FORMAT_RGBA8, RS2_FORMAT_BGR8, RS2_FORMAT_BGRA8 };
+
+    switch( source_format )
+    {
+    case RS2_FORMAT_YUYV:
+        formats.push_back( RS2_FORMAT_YUYV );
+        formats.push_back( RS2_FORMAT_Y16 );
+        formats.push_back( RS2_FORMAT_Y8 );
+        break;
+    case RS2_FORMAT_UYVY:
+        formats.push_back( RS2_FORMAT_UYVY );
+        break;
+    default:
+        throw std::runtime_error( "Format is not supported for mapping" );
+    }
+
+    return formats;
+}
+
+std::vector< tagged_profile > dds_rs_internal_data::get_profiles_tags( std::string product_id,
+                                                                       std::string product_line )
+{
+    std::vector<tagged_profile> tags;
+
+    if( product_id == RS405_PID )
```

Comment:
Why do it this way, out of curiosity -- why not add a virtual method to our d400 classes?

Reply:
I see, this is on the client...